### PR TITLE
[4.0] [com_contact] Missing parameterized queries, fixed double quoting

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds some missing parameterized queries and fixes possible double quoting.

### Testing Instructions

On multilingual site create some contacts with associations.
View contact list, check that associations appear like before.
View feature view in frontend.
Changes in `components/com_contact/Service/Route.php` can be tested by code review. Not sure if it's actually used anywhere.

### Expected result

Works.

### Documentation Changes Required

No.

### CODE REVIEW

File: components/com_contact/Model/FeaturedModel.php [hunk 1]
```diff
@@ -117,8 +117,7 @@ protected function getListQuery()
 			$query->bind(':published', $state, ParameterType::INTEGER);
 
 			// Filter by start and end dates.
-			$date = Factory::getDate();
-			$nowDate = $db->quote($date->toSql());
+			$nowDate = Factory::getDate()->toSql();
```
Comment (Quy) [inline components/com_contact/Model/FeaturedModel.php:120]: Remove double quoting on line 135?

Comment (SharkyKZ) [reply components/com_contact/Model/FeaturedModel.php:120]: Removed.

File: administrator/components/com_contact/Service/HTML/AdministratorService.php [hunk 2]
```diff
@@ -50,16 +51,23 @@ public function association($contactid)
...
+						$db->quoteName('lang_code'),
+						$db->quoteName('cat.title', 'category_title'),
+						$db->quoteName('l.image'),
+						$db->quoteName('l.title', 'language_title'),
+					]
+				)
+				->from($db->quoteName('#__contact_details', 'c'))
+				->join('LEFT', $db->quoteName('#__categories', 'cat'), $db->quoteName('cat.id') . ' = ' . $db->quoteName('c.catid'))
+				->join('LEFT', $db->quoteName('#__languages', 'l'), $db->quoteName('c.language') . ' = ' . $db->quoteName('l.lang_code'))
+				->whereIn($db->quoteName('c.id'), array_values($associations), ParameterType::STRING)
```
Comment (alikon) [inline administrator/components/com_contact/Service/HTML/AdministratorService.php]: that's weird the field `id` of `#__contact_details` table is a SERIAL at least on postgresql should be binded to an INTEGER imo

**Review by alikon — CHANGES_REQUESTED** [review]

File: components/com_contact/Model/FeaturedModel.php [hunk 3]
```diff
@@ -133,7 +132,7 @@ protected function getListQuery()
 		// Filter by language
 		if ($this->getState('filter.language'))
 		{
-			$language = [Factory::getLanguage()->getTag(), $db->quote('*')];
+			$language = [Factory::getLanguage()->getTag(), '*'];
 			$query->whereIn($db->quoteName('a.language'), $language);
```
Comment (alikon) [inline components/com_contact/Model/FeaturedModel.php]:
```suggestion
			$query->whereIn($db->quoteName('a.language'), $language, ParameterType::STRING);
```

**Review by alikon — CHANGES_REQUESTED** [review]

File: administrator/components/com_contact/Service/HTML/AdministratorService.php (hunk 2 above)
Comment (SharkyKZ) [reply administrator/components/com_contact/Service/HTML/AdministratorService.php]: Yes.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ded480ca3513ac57749779bf724c7fb83b41f263<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27012">issues.joomla.org/tracker/joomla-cms/27012</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on ded480ca3513ac57749779bf724c7fb83b41f263<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27012">issues.joomla.org/tracker/joomla-cms/27012</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27012">issues.joomla.org/tracker/joomla-cms/27012</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**